ESP8266/ESP8266.cpp

```diff
@@ -265,6 +321,10 @@ void ESP8266::_packet_handler()
 int32_t ESP8266::recv(int id, void *data, uint32_t amount)
 {
     while (true) {
+        _smutex.lock();
+        // Poll for inbound packets
+        _parser.process_oob();
```

Comment:
Why did this extra process call need to be added? Not saying it's wrong, just that it's changing the code flow a bit, and it makes me a little nervous.

Reply:
You are also ignoring errors from this `process_oob()` call.
Then, end of the loop, the same function is called but errors are checked.

You need to decide whether it needs to be called beginning of the loop, or at the end of the loop. And move it there (with error checks).

Reply:
Beginning it is and while-loop will be removed.